librz/core/cmd/cmd_print.c

```diff
@@ -5926,6 +5926,251 @@ static RzCmdStatus print_histogram_entropy(RzCore *core, int argc, const char **
+		// than if entopy goes above threshold printing rising entropy edge
+		// and if entropy goes below threshold printing falling entropy edge
+		if (resetFlag == 1 && data >= risingthreshold) {
+			// rising edge print
+			resetFlag = 0;
+			lastEdge = 1;
+			rz_strbuf_appendf(buf, "0x%08" PFMT64x " ", off);
+			rz_strbuf_appendf(buf, "%03x Rising entropy edge (%8lf)\n", i, data);
+		} else if (resetFlag == 1 && data <= fallingthreshold) {
+			// falling edge print
+			resetFlag = 0;
+			lastEdge = 0;
+			rz_strbuf_appendf(buf, "0x%08" PFMT64x " ", off);
+			rz_strbuf_appendf(buf, "%03x Falling entropy edge (%8lf)\n", i, data);
+		}
+	}
+}
+static RzCmdStatus print_rising_and_falling_entropy(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	double risingthreshold = 0.95;
+	double fallingthreshold = 0.85;
+	if (argc >= 3) {
+		risingthreshold = rz_num_get_float(core->num, argv[1]);
+		fallingthreshold = rz_num_get_float(core->num, argv[2]);
+	}
+	if (fallingthreshold > risingthreshold) {
+		RZ_LOG_ERROR("falling threshold is greater than rising threshold");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	if (risingthreshold > 1) {
+		RZ_LOG_ERROR("threshold can't be greater than 1");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	int nblocks = -1;
+	ut64 totalsize = UT64_MAX;
+	int skipblocks = -1;
+	CoreBlockRange *brange = calculate_blocks_range(core, 0, 0, totalsize, nblocks, skipblocks);
+	if (!brange) {
+		RZ_LOG_ERROR("Cannot calculate blocks range\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	double *data = calloc(1, 8 * brange->nblocks);
+	ut8 *tmp = malloc(brange->blocksize);
+	if (!tmp) {
+		RZ_LOG_ERROR("core: failed to malloc memory");
+		free(data);
+		free(brange);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	// bool resetFlag = 1;
+	// st8 lastEdge = 0;
+	RzStrBuf *buf = rz_strbuf_new("");
+	if (!buf) {
+		RZ_LOG_ERROR("core: failed to malloc memory");
+		free(data);
+		free(tmp);
+		free(brange);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	RzOutputMode mode = state->mode;
+	switch (mode) {
+	case RZ_OUTPUT_MODE_TABLE:
+		print_rising_and_falling_entropy_table(core, state, brange, tmp, fallingthreshold, risingthreshold);
+		break;
+	case RZ_OUTPUT_MODE_JSON:
+		print_rising_and_falling_entropy_JSON(core, state, brange, tmp, fallingthreshold, risingthreshold);
+		break;
+	case RZ_OUTPUT_MODE_QUIET:
+		print_rising_and_falling_entropy_quiet(core, buf, brange, tmp, fallingthreshold, risingthreshold);
+		break;
+	case RZ_OUTPUT_MODE_STANDARD:
+		print_rising_and_falling_entropy_standard(core, buf, brange, tmp, fallingthreshold, risingthreshold);
+		break;
+	case RZ_OUTPUT_MODE_LONG:
+		print_rising_and_falling_entropy_long(core, buf, brange, tmp, fallingthreshold, risingthreshold);
+		break;
+	default:
+		rz_warn_if_reached();
+		rz_strbuf_free(buf);
+		free(tmp);
+		free(data);
+		free(brange);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	free(tmp);
+	if (mode == RZ_OUTPUT_MODE_STANDARD || mode == RZ_OUTPUT_MODE_QUIET || mode == RZ_OUTPUT_MODE_LONG) {
```

Comment:
This is unnecessary when `RzCmdStateOutput` is used

Reply:
Use `rz_cmd_state_init()` and `rz_cmd_state_fini()` or something like that - search other handlers